### PR TITLE
move some info logs to trace, add a detailed recipe state test

### PR DESCRIPTION
**Pull request by rbaynes**
Extending the tests that @rwdavis513 started

### CODE REVIEW

File: nodes/recipe_handler.py [hunk 1]
```diff
@@ -71,14 +71,16 @@ def trace(msg, *args):
 
 
 #------------------------------------------------------------------------------
+# Returns the current 'state', an unordered dictionary of the current value
```
Comment (gordonbrander) [inline nodes/recipe_handler.py]: This should be part of the docstring for the function below. Keeping function documentation in docstrings allows them to be accessed computationally, so you can do things like

- generate documentation from comments
- automatically run tests in comments
- `help(func_name)` .

File: nodes/recipe_handler.py [hunk 2]
```diff
@@ -71,14 +71,16 @@ def trace(msg, *args):
...
+# of each variable at now_time.
 def interpret_simple_recipe(recipe, start_time, now_time):
     """
     Produces a tuple of ``(variable, value)`` pairs by building up
     a recipe state from walking through the recipe keyframes
     """
     _id = recipe["_id"]
     operations = recipe["operations"]
-    rospy.loginfo(operations)
+    #trace("recipe_handler: operations: %s", operations)
```
Comment (gordonbrander) [inline nodes/recipe_handler.py]: commented out?

File: test/test_nodes/test_recipe_handler.py [hunk 3]
```diff
@@ -43,17 +69,120 @@ def test_recipe_interpreter(self):
...
+        assert setpoints[6][0] == 'one', \
+              'setpoints[6][0]=%s but should be one' % setpoints[6][0]
+
+        # Test for recipe in process at last time change (180 secs in)
+        now_time = time()
+        start_time = now_time - 180
+        setpoints = interpret_simple_recipe(MOCK_RECIPE_B, start_time, now_time)
+        assert len(setpoints) == 8, \
+            'len(setpoints)=%d should be 8' % len(setpoints)
+        #assert False, '%s, %s, %s, %s, %s, %s, %s, %s' % (setpoints[0][0], setpoints[1][0], setpoints[2][0], setpoints[3][0], setpoints[4][0], setpoints[5][0], setpoints[6][0], setpoints[7][0])
```
Comment (gordonbrander) [inline test/test_nodes/test_recipe_handler.py]: commented out?

File: test/test_nodes/test_recipe_handler.py [hunk 4]
```diff
@@ -43,17 +69,120 @@ def test_recipe_interpreter(self):
...
         setpoints = interpret_simple_recipe(MOCK_RECIPE_A, start_time, now_time)
-        assert len(setpoints) == 2
+        assert len(setpoints) == 2, 'len(setpoints)=%d' % len(setpoints)
         # Test for completed recipe
         start_time = now_time - 300
         setpoints = interpret_simple_recipe(MOCK_RECIPE_A, start_time, now_time)
-        assert setpoints[0][0] == 'recipe_end'
+        assert setpoints[0][0] == 'recipe_end', 'recipe_end=' % setpoints[0][0]
+
+    #--------------------------------------------------------------------------
```
Comment (gordonbrander) [inline test/test_nodes/test_recipe_handler.py]: No need for these separators, I think?

File: test/test_nodes/test_recipe_handler.py [hunk 5]
```diff
@@ -43,17 +69,120 @@ def test_recipe_interpreter(self):
...
+        assert setpoints[4][0] == 'one', \
+              'setpoints[4][0]=%s but should be one' % setpoints[4][0]
+
+        # Test for recipe in process at next change (120 secs in)
+        now_time = time()
+        start_time = now_time - 120
+        setpoints = interpret_simple_recipe(MOCK_RECIPE_B, start_time, now_time)
+        assert len(setpoints) == 7, \
+            'len(setpoints)=%d should be 7' % len(setpoints)
+        #assert False, '%s, %s, %s, %s, %s, %s, %s' % (setpoints[0][0], setpoints[1][0], setpoints[2][0], setpoints[3][0], setpoints[4][0], setpoints[5][0], setpoints[6][0])
```
Comment (gordonbrander) [inline test/test_nodes/test_recipe_handler.py]: commented out?

File: test/test_nodes/test_recipe_handler.py (hunk 4 above)
Comment (Spaghet) [inline test/test_nodes/test_recipe_handler.py]: Agree

**Comment by rwdavis513** [comment]
Merged these changes into [PR266](https://github.com/OpenAgInitiative/openag_brain/pull/266)

Closing this Pull Request as it is covered by the other one which also fixes the runtime issues on the current master branch.

**Pull request closed without merging**